src/gConsent/manage/getAccessGrantAll.ts

```diff
@@ -97,20 +103,39 @@ async function internal_getAccessGrantAll(
 
   const specifiedModes = accessToResourceAccessModeArray(params.access ?? {});
 
+  const statusShorthand = params.status ?? "granted";
+  const statusFull: string[] = [];
+
+  if (["granted", "all"].includes(statusShorthand)) {
+    statusFull.push(GC_CONSENT_STATUS_EXPLICITLY_GIVEN);
+  }
+
+  if (["denied", "all"].includes(statusShorthand)) {
+    statusFull.push(GC_CONSENT_STATUS_DENIED);
+  }
+
   const vcShapes: RecursivePartial<BaseGrantBody & VerifiableCredential>[] =
-    ancestorUrls.map((url) => ({
-      "@context": [CONTEXT_VC_W3C, CONTEXT_ESS_DEFAULT],
-      type: [CREDENTIAL_TYPE_ACCESS_GRANT, CREDENTIAL_TYPE_BASE],
-      credentialSubject: {
-        providedConsent: {
-          hasStatus: GC_CONSENT_STATUS_EXPLICITLY_GIVEN,
-          forPersonalData: url ? [url.href] : undefined,
-          forPurpose: params.purpose,
-          isProvidedTo: params.requestor,
-          mode: specifiedModes.length > 0 ? specifiedModes : undefined,
-        },
-      },
-    }));
+    statusFull.flatMap(
```

Comment:
Instead of querying for both denied and approved grants if `all` is specified, I would have expected we just don't filter at all on the status. This would result in not having a value in the VC shape for the status, instead of sending two queries, one for each value.

Eventually, we'll want this to be the bahavior on `undefined`, but for backwards compatibility `undefined` maps to approved. So the possible cases are: 
- approved or undefined => status is set in the shape to approved
- denied => status is set to denied
- all => status isn't set, and all values for the status will be accepted on the Holder side.

Does this make sense, or am I missing something?

Reply:
I've tried that over [here](https://github.com/inrupt/solid-client-access-grants-js/pull/756) and it seems to work.

My main concern was that there would be noise from other things like access requests because we are both setting `hasStatus` but we are also being less restrictive in the `type` field.

No extra noise appears to be turning in those e2e tests so I think it is fine to merge that PR into this one and proceed if you're understanding of the VC service implementation supports this not being an issue.

Reply:
There are other differences between Access grants and Access Requests beyond the `status`, the predicates used in the `credentialSubject` will differ, with `providedConsent` in Grants and `hasConsent` (or something similar, this is from top of head) in Requests. All this to say, ever wihtout the status being set, no Access Request will match this request, so I think we should be good there